### PR TITLE
Eliminate internal, ancient glob() and fnmatch() copies

### PR DESCRIPTION
**Pull request by pmatilai**
glibc >= 2.27 thankfully reverted the braindead behavior of glob() and fnmatch() not returning dangling symlinks, so we can finally drop our internal copy of the ancient (and buggy) glibc 2.1 functions. More details in the commit message.

### CODE REVIEW

File: rpmio/rpmglob.c [hunk 1]
```diff
@@ -30,84 +25,15 @@
...
-#ifndef HAVE_MEMPCPY
-static void * mempcpy(void *dest, const void *src, size_t n)
-{
-    return (char *) memcpy(dest, src, n) + n;
-}
-#endif
-
 /* Find the end of the sub-pattern in a brace expression.  We define
    this as an inline function if the compiler permits.  */
 static inline const char *next_brace_sub(const char *begin)
```
Comment (dmnks) [inline rpmio/rpmglob.c:39]: Shouldn't this line be removed too?

Comment (dmnks) [reply rpmio/rpmglob.c:39]: Nah, nvm, my bad, it's used by `__glob_pattern_p`.

**Comment by dmnks** [comment]
LGTM. This is much welcome, and will also make my life easier in #1903, so thanks, I guess? :smile:

**Comment by pmatilai** [comment]
Hmm, the downside of this is that we're relying on multiple flags (GLOB_BRACE, GLOB_TILDE, GLOB_DIRONLY) that are glibc specific extensions. This was not an issue when we bundled said implementation but we may see some portability complaints now, the POSIX specification is very barebones compared to glibc: https://pubs.opengroup.org/onlinepubs/9699919799/functions/glob.html

**Comment by dmnks** [comment]
Yeah... I came to the same realization today when I looked at the man page again and noticed that e.g. `GLOB_BRACE` was indeed a glibc extension, but I thought at first that it would be fine if we just handled that ourselves in rpmGlob() by expanding it and calling glob() multiple times - except that the logic isn't that simple, plus there are other such flags as you said... That sucks.

**Comment by pmatilai** [comment]
Portability sucks, for the developers. If this becomes an actual issue, we'll look into integrating gnulib rather than maintaining our own copy.

**Pull request merged**